bundles/alpha.model/src/alpha/model/util/Face.xtend

```diff
@@ -115,8 +118,13 @@ class Face {
 	
 	/** Enumerates the set of all possible label combinations. */
 	static def enumerateAllPossibleLabelings(int nbFacets, boolean includeNeg) {
-		val labels = includeNeg ? #[Label.POS, Label.ZERO, Label.NEG] : #[Label.POS, Label.ZERO] 
-		return labels.permutations(nbFacets)
+		val labels = includeNeg ? #[Label.ZERO, Label.POS, Label.NEG] : #[Label.POS, Label.ZERO]
+		return labels.permutations(nbFacets).filter[isValid]
+	}
+	
+	/** Returns True if labeling contains at least one POS and NEG label, or false otherwise */
+	static def isValid(Label... labeling) {
+		return labeling.exists[it == Label.POS] && labeling.exists[it == Label.NEG]
```

Comment:
This is an attempt to exclude searching labelings that we know are impossible. And now we check all combos with ZERO faces first, since this PR adds detection and removal of identical answers which only happens when most of the facets are ZERO faces. This is an attempt to force the exploration to stumble upon such scenarios sooner.